### PR TITLE
[pvr] fix setting initial recording size value from addon

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

The initial value for recording size loaded by the addon was not checked for validity thereby allowing negative values which are not currently permitted.

## Motivation and Context

Fix the issue.

## How Has This Been Tested?

Mac OSX

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
This should fix any negative values being passed by the addon causing an issue.

@emveepee FYI

File: xbmc/pvr/recordings/PVRRecording.cpp [hunk 1]
```diff
@@ -97,7 +97,8 @@ CPVRRecording::CPVRRecording(const PVR_RECORDING& recording, unsigned int iClien
   if (strlen(recording.strFirstAired) > 0)
     m_firstAired.SetFromW3CDateTime(recording.strFirstAired);
   m_iFlags = recording.iFlags;
-  m_sizeInBytes = recording.sizeInBytes;
+  if (recording.sizeInBytes > 0)
```
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecording.cpp:100]: Strictly speaking >= 0. Theoretically one could change the init value of the recording size member to something <> 0, then your code breaks.

**Comment by ksooo** [comment]
By the way. CPPVRecording::SetSizeInBytes is unused and should be removed.

File: xbmc/pvr/recordings/PVRRecording.cpp (hunk 1 above)
Comment (phunkyfish) [reply xbmc/pvr/recordings/PVRRecording.cpp:100]: Good point.

**Comment by phunkyfish** [comment]
Should be correct now. Unused function also removed.

**Comment by ksooo** [comment]
>  [pvr] remove unused GetSizeInBytes function call

 Nope. You removed the setter, not the getter.

**Comment by phunkyfish** [comment]
Ok, should be right now.

**Comment by ksooo** [comment]
I think we should update the API documentation to mention the special -1 value meaning, maybe even introduce a define for that.

**Comment by phunkyfish** [comment]
Fair, as the work is underway on the C++ API we should add it to that PR.

**Comment by ksooo** [comment]
Okay.

**Review by ksooo — APPROVED** [review]
looks good now

**Comment by ksooo** [comment]
> Fair, as the work is underway on the C++ API we should add it to that PR.

 If your time allows, could you take care of that?

**Comment by emveepee** [comment]
> 
> 
> This should fix any negative values being passed by the addon causing an issue.
> 
> @emveepee FYI

In the issue I indentified the -1 wasn't coming from the addon but I still need to change my code to handle directory access without read permission which would be undefined if it was encountered.

**Comment by phunkyfish** [comment]
> > Fair, as the work is underway on the C++ API we should add it to that PR.
> 
> If your time allows, could you take care of that?

Of course.

**Review by AlwinEsch — APPROVED** [review]

**Pull request merged**